app/src/main/java/com/globant/counter/android/mvp/presenter/CalculatePresenter.java

```diff
@@ -0,0 +1,66 @@
+package com.globant.counter.android.mvp.presenter;
+
+import com.globant.counter.android.R;
+import com.globant.counter.android.mvp.model.CalculateModel;
+import com.globant.counter.android.mvp.view.CalculateView;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author s.ruiz
+ */
+
+public class CalculatePresenter {
+    private CalculateView view;
+    private CalculateModel model;
+
+    public CalculatePresenter(CalculateView view) {
+        this.view = view;
+        model = new CalculateModel();
+    }
+
+    public void setView(CalculateView view){
+        if (view != null) {
+            this.view = view;
+        }
+    }
+
+    public void onEqualPressed() {
+        if (view != null) {
```

Comment:
This condition is invalid. Because you passed the object in the constructor. If you like, you could add @NotNull annotation in the constructor.